rq/worker.py

```diff
@@ -635,6 +637,8 @@ def perform_job(self, job, queue):
         else:
             self.log.warning('Result will never expire, clean up result key manually')
 
+        pop_connection()
```

Comment:
This should be called in a `finally:` block, to make sure it's always called, even in case of an exception.